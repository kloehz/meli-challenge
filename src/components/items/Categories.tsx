interface ICategories {
    categories: string[]
}

export const Categories = ({categories}: ICategories) => {
    return (
        <div className="categories-container">
            {
                categories.map((category) => {
                    return (
                        <h5
                            key={category}
                            className="category"
                        >
                            {category}
                        </h5>
                    )
                })
            }
        </div>
    )
}
