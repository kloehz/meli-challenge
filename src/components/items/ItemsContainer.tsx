import useCurrency from '../../hooks/useCurrency';
import { Item } from '../../types/getItems';
import { useHistory } from 'react-router-dom';
import { Prices } from '../prices/Prices';

export const ItemsContainer = ({id, picture, title, price, seller_address}: Item) => {

    const history = useHistory();

    const handleClick = () => {
            history.push({pathname: `/items/${id}`});
    };

    const parsedPrice = useCurrency({...price});

    return (
        <div className="product-container" onClick={handleClick}>
            <div className="item-image-container">
                <img src={picture} />
            </div>
        <div className="title-container">
            <Prices price={parsedPrice} style='item' />
                <h4>{title}</h4>
            </div>
            <div className="seller-address-container">
                <h4 className="seller-address">{seller_address}</h4>
            </div>
        </div>
    )
}
