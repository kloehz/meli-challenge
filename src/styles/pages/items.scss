.items-container {
    width: 80vw;
    margin: 0 auto;
}

.product-container {
    display: flex;
    background-color: white;
    margin-bottom: 4px;
}