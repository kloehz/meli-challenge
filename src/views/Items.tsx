import { useLocation } from 'react-router-dom';
import * as queryString from "query-string";
import { IGetItems } from '../types/getItems';
import { useCapitalize } from '../hooks/useCapitalize';
import { ItemsContainer } from '../components/items/ItemsContainer';
import { Categories } from '../components/items/Categories';
import { Seo } from '../components/seo/Seo';
import { NotResuts } from '../components/items/NotResuts';


export const Items = () => {
        
    const location = useLocation<IGetItems>();
    const { q: title } =  queryString.parse(location.search);
    const { items, categories } = location.state;
    const capitalizedTitle = useCapitalize(title.toString());

    return (
        <>
            <Seo title={capitalizedTitle} />
            <div className="items-page">
                <div className="items-container">
                    {
                        categories.length > 0 && <Categories categories={categories}/>
                    }
                    {
                        items.length ? items.map((item) => {
                            return <ItemsContainer key={item.id} {...item} />
                        })
                        : <NotResuts />
                    }
                </div>
            </div>
        </>
    )
}
