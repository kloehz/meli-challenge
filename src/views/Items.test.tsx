import { shallow } from "enzyme"
import { Items } from "./Items"
import { Categories } from "../components/items/Categories"
import { ItemsContainer } from "../components/items/ItemsContainer";

jest.mock('react-router-dom', () => ({
    // @ts-ignore
    ...jest.requireActual("react-router-dom"),
    useLocation: jest.fn().mockReturnValue({
        pathname: '',
        search: '?q=vasos',
        hash: '',
        state: {"author":{"name":"Guido Leonel","lastname":"Cotelesso"},"categories":["Hogar, Muebles y Jardín","Bazar y Cocina","Vajilla y Artículos de Servir","Vajilla","Vasos"],"items":[{"id":"MLA912120163","title":"Set X6 Vasos Vidrio Color Trago Largo 465 Cc Pulsar Bormioli","price":{"currency":"ARS","amount":1349,"decimals":10},"picture":"http://http2.mlstatic.com/D_625911-MLA45228911173_032021-O.jpg","condition":"Nuevo","free_shipping":false,"seller_address":"Santa Fe"},{"id":"MLA854588801","title":"Vaso Copon Copa Sin Tallo Rigolleau Gourmet 450ml X 6","price":{"currency":"ARS","amount":895,"decimals":0},"picture":"http://http2.mlstatic.com/D_676696-MLA43737946576_102020-O.jpg","condition":"Nuevo","free_shipping":false,"seller_address":"Capital Federal"},{"id":"MLA876626363","title":"Vaso Vidrio Copa S/tallo Copón Vino Gourmet Rigolleau X6","price":{"currency":"ARS","amount":850,"decimals":0},"picture":"http://http2.mlstatic.com/D_914976-MLA46636137693_072021-O.jpg","condition":"Nuevo","free_shipping":false,"seller_address":"Capital Federal"},{"id":"MLA877200255","title":"Vaso Pampa Durax De Vidrio 400ml Agua Jugo X 12 Unid Oferta","price":{"currency":"ARS","amount":869,"decimals":0},"picture":"http://http2.mlstatic.com/D_923689-MLA43399324060_092020-O.jpg","condition":"Nuevo","free_shipping":false,"seller_address":"Capital Federal"}]},
        key: 'oaent654',
    })
}))

describe('Items Page test', () => {
      
    const wrapper = shallow(<Items />
    );

    test('should be match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should be render categories', () => {
        expect(wrapper.containsMatchingElement(<Categories categories={["Hogar, Muebles y Jardín","Bazar y Cocina","Vajilla y Artículos de Servir","Vajilla","Vasos"]} />))
        .toEqual(true);
    });
    

    // test('should be not render categories', () => {
    //     jest.mock('react-router-dom', () => ({
    //         // @ts-ignore
    //         ...jest.requireActual("react-router-dom"),
    //         useLocation: jest.fn().mockReturnValue({
    //             pathname: '',
    //             search: '?q=vasos',
    //             hash: '',
    //             state: {"author":{"name":"Guido Leonel","lastname":"Cotelesso"},"categories":[],"items":[{"id":"MLA912120163","title":"Set X6 Vasos Vidrio Color Trago Largo 465 Cc Pulsar Bormioli","price":{"currency":"ARS","amount":1349,"decimals":10},"picture":"http://http2.mlstatic.com/D_625911-MLA45228911173_032021-O.jpg","condition":"Nuevo","free_shipping":false,"seller_address":"Santa Fe"},{"id":"MLA854588801","title":"Vaso Copon Copa Sin Tallo Rigolleau Gourmet 450ml X 6","price":{"currency":"ARS","amount":895,"decimals":0},"picture":"http://http2.mlstatic.com/D_676696-MLA43737946576_102020-O.jpg","condition":"Nuevo","free_shipping":false,"seller_address":"Capital Federal"},{"id":"MLA876626363","title":"Vaso Vidrio Copa S/tallo Copón Vino Gourmet Rigolleau X6","price":{"currency":"ARS","amount":850,"decimals":0},"picture":"http://http2.mlstatic.com/D_914976-MLA46636137693_072021-O.jpg","condition":"Nuevo","free_shipping":false,"seller_address":"Capital Federal"},{"id":"MLA877200255","title":"Vaso Pampa Durax De Vidrio 400ml Agua Jugo X 12 Unid Oferta","price":{"currency":"ARS","amount":869,"decimals":0},"picture":"http://http2.mlstatic.com/D_923689-MLA43399324060_092020-O.jpg","condition":"Nuevo","free_shipping":false,"seller_address":"Capital Federal"}]},
    //             key: 'oaent654',
    //         })
    //     }));

    //     expect(wrapper.containsMatchingElement(<Categories categories={[]} />))
    //     .toEqual(true);
    // })
    
})


