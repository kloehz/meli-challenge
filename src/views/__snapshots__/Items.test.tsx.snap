// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Items Page test should be match snapshot 1`] = `ShallowWrapper {}`;
